consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/BlockTimer.java

```diff
@@ -76,16 +84,36 @@ public synchronized boolean isRunning() {
    */
   public synchronized void startTimer(
       final ConsensusRoundIdentifier round, final BlockHeader chainHeadHeader) {
-    cancelTimer();
-
-    final long now = clock.millis();
 
     // absolute time when the timer is supposed to expire
-    final int blockPeriodSeconds =
+    final int currentBlockPeriodSeconds =
         forksSchedule.getFork(round.getSequenceNumber()).getValue().getBlockPeriodSeconds();
-    final long minimumTimeBetweenBlocksMillis = blockPeriodSeconds * 1000L;
+    final long minimumTimeBetweenBlocksMillis = currentBlockPeriodSeconds * 1000L;
     final long expiryTime = chainHeadHeader.getTimestamp() * 1_000 + minimumTimeBetweenBlocksMillis;
 
+    setBlockTimes(round, false);
+
+    startTimer(round, expiryTime);
+  }
+
+  public synchronized void startEmptyBlockTimer(
+          final ConsensusRoundIdentifier round, final BlockHeader chainHeadHeader) {
+
+    // absolute time when the timer is supposed to expire
+    final int currentBlockPeriodSeconds =
+            forksSchedule.getFork(round.getSequenceNumber()).getValue().getEmptyBlockPeriodSeconds();
+    final long minimumTimeBetweenBlocksMillis = currentBlockPeriodSeconds * 1000L;
+    final long expiryTime = chainHeadHeader.getTimestamp() * 1_000 + minimumTimeBetweenBlocksMillis;
+
+    setBlockTimes(round, true);
+
+    startTimer(round, expiryTime);
+  }
+
+  public synchronized void startTimer(final ConsensusRoundIdentifier round, final long expiryTime) {
```

Comment:
```suggestion
  private synchronized void startTimer(final ConsensusRoundIdentifier round, final long expiryTime) {
```

Reply:
I don't know about this one, it is indeed being called fron inside the class, but I wonder of at some point it may needed to be public...

Reply:
Whoever needs it to be public in the future can make that change then :) Think we should keep it private for now.

Reply:
Will do...